core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -61,19 +86,51 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
+      throw e.toThrift();
+    } catch (IOException e) {
+      LOG.debug("I/O Error: {}", callable, e);
```

Comment:
Resolved offline, this isn't our current contract so I will just use `Exit (Error)`.